src/ScheduledTasks/Client/ScheduledTaskClientImpl.cs

```diff
@@ -0,0 +1,180 @@
+    {
+        Check.NotNull(creationOptions, nameof(creationOptions));
+        this.logger.ClientCreatingSchedule(creationOptions);
+
+        try
+        {
+            // Create schedule client instance
+            ScheduleClient scheduleClient = new ScheduleClientImpl(this.durableTaskClient, creationOptions.ScheduleId, this.logger);
+
+            // Create the schedule using the client
+            await scheduleClient.CreateAsync(creationOptions, cancellation);
+
+            return scheduleClient;
+        }
+        catch (OperationCanceledException) when (cancellation.IsCancellationRequested)
+        {
+            // the operation was cancelled as requested. No need to log this.
+            throw;
+        }
+        catch (Exception ex)
+        {
+            this.logger.ClientError(
+                nameof(this.CreateScheduleAsync),
+                creationOptions.ScheduleId,
+                ex);
+
+            throw;
+        }
+    }
+
+    /// <inheritdoc/>
+    public override async Task<ScheduleDescription?> GetScheduleAsync(string scheduleId, CancellationToken cancellation = default)
+    {
+        Check.NotNullOrEmpty(scheduleId, nameof(scheduleId));
+
+        try
+        {
+            // Get schedule client first
+            ScheduleClient scheduleClient = this.GetScheduleClient(scheduleId);
+
+            // Call DescribeAsync which handles all the entity state mapping
+            return await scheduleClient.DescribeAsync(cancellation);
+        }
+        catch (ScheduleNotFoundException)
+        {
+            // Return null if schedule not found
+            return null;
+        }
+        catch (OperationCanceledException) when (cancellation.IsCancellationRequested)
+        {
+            // the operation was cancelled as requested. No need to log this.
+            throw;
+        }
+        catch (Exception ex)
+        {
+            this.logger.ClientError(
+                nameof(this.GetScheduleAsync),
+                scheduleId,
+                ex);
+
+            throw;
+        }
+    }
+
+    /// <inheritdoc/>
+    public override ScheduleClient GetScheduleClient(string scheduleId)
+    {
+        Check.NotNullOrEmpty(scheduleId, nameof(scheduleId));
+        return new ScheduleClientImpl(this.durableTaskClient, scheduleId, this.logger);
+    }
+
+    /// <inheritdoc/>
+    public override AsyncPageable<ScheduleDescription> ListSchedulesAsync(ScheduleQuery? filter = null)
+    {
+        // Create an async pageable using the Pageable.Create helper
+        return Pageable.Create(async (continuationToken, pageSize, cancellation) =>
+        {
+            try
+            {
+                // TODO: map to entity query last modified from/to filters
+                EntityQuery query = new EntityQuery
+                {
+                    InstanceIdStartsWith = filter?.ScheduleIdPrefix ?? nameof(Schedule),
```

Comment:
We need to always specify a filter, otherwise this returns ALL of the user-defined entities as well. There could be millions of those and which would completely break this query.

This should work:
```csharp
InstanceIdStartsWith = $"@{nameof(Scheduler)}@{filter?.ScheduleIdPrefix ?? string.Empty}"
```